lib/types.ts

```diff
@@ -52,9 +52,9 @@ export interface ScannedProjectCustom {
 }
 
 export interface DepTreeDep {
-  name: string;
-  version: string;
-  dependencies: {
+  name?: string;
```

Comment:
the CLI doesn't know what it's getting or what plugin it's calling necessarily, so it doesn't mark these fields as mandatory.
the Plugin does know what it aims to return, so marking them as optional would be a mistake. I think we should definitely remove this commit

Reply:
But how does scanning a Node app produce `packageFormatVersion` or a `targetOS` object? How is `docker` also required for such results?

Reply:
I'm also following the interfaces in the `lockfile-parser`, otherwise we get a significant mismatch

Reply:
it's not, we need to distinguish the type for the OS scan and the "app" scan.
the OS scan must have these fields, the "app" scan should have... whatever it has to have (for you to discover 😛 )
at least I think it's better to distinguish these two types...